src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesTaskLauncher.java

```diff
@@ -390,7 +391,7 @@ private void deleteJob(String id) {
 				.withLabel(SPRING_APP_KEY, id);
 
 		if (jobsToDelete == null || ObjectUtils.isEmpty(jobsToDelete.list().getItems())) {
-			throw new IllegalStateException(String.format("Cannot delete job for task \"%s\" (reason: job does not exist)", id));
```

Comment:
You will need to update the tests as well - you can see what I added in the previous commit https://github.com/spring-cloud/spring-cloud-deployer-kubernetes/commit/c89dd43ba5b7210165ed1278caeaaba712d7bb05

Reply:
Updated.
Thanks for catching that, I looked at the original commit and missed the added tests.